tests/testCondition.py

```diff
@@ -0,0 +1,302 @@
+        self.assertEqual(cond.params, ("tunnel", ))
+        self.assertTrue(cond.test({"tunnel": "yes"}))
+        self.assertTrue(cond.test({"tunnel": "maybe"}))
+        self.assertTrue(cond.test({"tunnel": "+1"}))
+        self.assertFalse(cond.test({"highway": "secondary"}))
+
+        cond:Condition = parseCondition("building\t")
+        self.assertEqual(cond.type, "set")
+        self.assertEqual(cond.params, ("building", ))
+        self.assertTrue(cond.test({"building": "yes"}))
+        self.assertTrue(cond.test({"building": "apartment"}))
+        self.assertTrue(cond.test({"building": "1"}))
+        self.assertFalse(cond.test({"highway": "secondary"}))
+        self.assertFalse(cond.test({"building:part": "yes"}))
+
+        cond:Condition = parseCondition(" addr:housenumber ")
+        self.assertEqual(cond.type, "set")
+        self.assertEqual(cond.params, ("addr:housenumber", ))
+        self.assertTrue(cond.test({"addr:housenumber": "1"}))
+        self.assertTrue(cond.test({"addr:housenumber": "yes"}))
+        self.assertFalse(cond.test({"highway": "secondary"}))
+        self.assertFalse(cond.test({"addr:street": "Baker st"}))
+
+        cond:Condition = parseCondition(" some-tag ")
+        self.assertEqual(cond.type, "set")
+        self.assertEqual(cond.params, ("some-tag", ))
+        self.assertTrue(cond.test({"some-tag": "1"}))
+        self.assertTrue(cond.test({"some-tag": "yes"}))
+        self.assertFalse(cond.test({"highway": "secondary"}))
+        self.assertFalse(cond.test({"some": "tag"}))
+
+    def test_parser_unset(self):
+        cond:Condition = parseCondition("!tunnel")
+        self.assertEqual(cond.type, "unset")
+        self.assertEqual(cond.params, ("tunnel", ))
+        self.assertTrue(cond.test({"capital": "1"}))
+        self.assertFalse(cond.test({"tunnel": "yes"}))
+        self.assertFalse(cond.test({"tunnel": "no"}))
+
+        cond:Condition = parseCondition("\t!name  ")
+        self.assertEqual(cond.type, "unset")
+        self.assertEqual(cond.params, ("name", ))
+        self.assertTrue(cond.test({"capital": "1"}))
+        self.assertTrue(cond.test({"int_name": "1"}))
+        self.assertFalse(cond.test({"name": "London"}))
+
+    def test_parser_false(self):
+        """ Test conditions in format some_tag = no
+            Note that such conditions are not used by Organic Maps styles.
+        """
+        cond:Condition = parseCondition("access=no")
+        self.assertEqual(cond.type, "false")
+        self.assertEqual(cond.params, ("access", ))
+        #self.assertTrue(cond.test({"access": "no"}))      # test is not implemented for `false` condition
+        #self.assertTrue(cond.test({"access": "private"})) # test is not implemented for `false` condition
+        self.assertFalse(cond.test({"tunnel": "yes"}))
+
+    def test_parser_invTrue(self):
+        """ Test conditions in format [!some_tag?] It works the same way as [some_tag != yes]
```

Comment:
we use either [some_tag?] (which means any value except `no`)
or [!some_tag] which means absence or =no

[!some_tag?] is very strange :)

Reply:
I mentioned in comment that this type of conditions is never used in OrganicMaps styles. Another candidate for removal.